### PR TITLE
[Sentry-issue] Fix issue with undefined exerciseCategories

### PR DESCRIPTION
**Pull request by MerlinMehmed**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation and Context
This PR solved this Sentry issue: [3096](https://sentry.ase.in.tum.de/organizations/artemis/issues/3096)

### Description

I added a check if the exerciseCategories list is defined before pushing items to it, if not a new one is initialized.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Open a course and go to Exercises
4. Choose Import Programming Exercise and proceed to the import screen
5. Open the console in the Developer tools of the browser (by F12 usually)
6. Add category (different from the one of the imported exercise) and check that there are no errors in the console
7. Fill in the rest of the mandatory data and finish the Import of the exercise
8. Check that the category you have added are available

### CODE REVIEW

**Review by FuchsDominik — APPROVED** [review]
Changes look like they definitely fix the issue

**Review by rs-maker — APPROVED** [review]
Works fine on TS3, no errors in console and code makes sense.

**Review by SimonEntholzer — APPROVED** [review]
LGTM

**Review by jhainzlschmid — APPROVED** [review]
Looks good to me

File: src/main/webapp/app/shared/category-selector/category-selector.component.ts [hunk 1]
```diff
@@ -74,7 +74,11 @@ export class CategorySelectorComponent implements OnChanges {
             const randomIndex = Math.floor(Math.random() * this.categoryColors.length);
             exerciseCategory.color = this.categoryColors[randomIndex];
         }
-        this.exerciseCategories.push(exerciseCategory);
+        if (this.exerciseCategories) {
+            this.exerciseCategories.push(exerciseCategory);
+        } else {
+            this.exerciseCategories = [exerciseCategory];
+        }
```
Comment (nicolasruscher) [inline src/main/webapp/app/shared/category-selector/category-selector.component.ts:81]:
```suggestion
       this.exerciseCategories = this.exerciseCategories ? this.exerciseCategories.push(exerciseCategory): [exerciseCategory];
```

**Review by nicolasruscher — APPROVED** [review]
I guess you could do it as a one liner (see my suggestion), but that's not necessary.

**Pull request merged**